app/services/datastream_downloader.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+# Service for dowloading (downstream) datastream files
+class DatastreamDownloader
+  def initialize(ssgs = nil)
+    @ssgs = ssgs || default_supported_ssgs
+  end
+
+  def default_supported_ssgs
+    ::SupportedSsg.available_upstream
+                  .sort_by(&:comparable_version)
+                  .reverse
+  end
+
+  def download_datastreams
+    Dir.mktmpdir do |tmpdir|
+      files = datastream_urls.map do |label, url|
+        filepath = File.join(tmpdir, label)
+        File.write(filepath, SafeDownloader.download(url))
+        filepath
+      end
+      yield files
+    end
+  end
+
+  def datastream_urls
+    @ssgs.map do |ssg|
+      [ssg.id, datastream_url(ssg)]
+    end
+  end
+
+  def datastream_url(ssg)
+    [
+      Settings.compliance_ssg_url,
+      'datastreams',
+      "rhel#{ssg.os_major_version}",
+      ssg.package,
+      "ssg-rhel#{ssg.os_major_version}-ds.xml"
+    ].join('/')
```

Comment:
@akofink Here is the draft path spec for the downstream datastreams. 

The path here includes a `package` name. I've expected to have all previous datastreams present on the compliance-ssg server. Using just the version (w/o the release part) might be insufficient in case of a backport.

Reply:
Sounds great! We should definitely use the entire package name IMO with the release, since sometimes only the release changes. For upstream syncing that didn't matter, but now we'll have to respond to release bumps to get the new content.

Reply:
https://ruby-doc.org/stdlib-2.5.1/libdoc/uri/rdoc/URI/HTTP.html

Reply:
:laughing: fair point. I'll update it.

Reply:
@skateman well, check the safe_downloader.rb. This path will eventually be safely parsed with the URI module. I'm not sure there's any benefit in also doing it here.